frontend/src/pages/ChangeDepositPage.tsx

```diff
@@ -0,0 +1,79 @@
+import {useNavigate, useParams} from "react-router-dom";
+import {ITokenConfig} from "../models/Connection";
+import React, {ChangeEvent, FormEvent, useContext, useEffect, useMemo, useState} from "react";
+import {AuthContext} from "../context/AuthProvider";
+import {IDepositController} from "../models/Deposit";
+import DepositController from "../controllers/DepositController";
+import {DatePicker, LocalizationProvider} from "@mui/lab";
+import AdapterDateFns from "@mui/lab/AdapterDateFns";
+import {TextField} from "@mui/material";
+import './ChangeDepositPage.scss'
+
+export default function ChangeDepositPage() {
+    const params = useParams()
+    const navigate = useNavigate()
+    const depositId = params.depositId
+    const config: ITokenConfig | undefined = useContext(AuthContext).config
+    const controller: IDepositController = useMemo(() => DepositController(config), [config])
+    const [description, setDescription] = useState<string>("")
+    const [amount, setAmount] = useState<number>(0)
+    const [date, setDate] = useState<Date>(new Date(Date.now()))
+
+    useEffect(() => {
+        if (depositId === undefined) navigate("/")
+        controller.getDeposit(depositId!).then(deposit => {
+            setDescription(deposit.description)
+            setAmount(deposit.amount)
+            setDate(deposit.depositDate)
+        })
+    }, [controller, depositId, navigate])
+
+    const changeDeposit = (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        if (description.length && amount && date) {
+            const depositDate: Date = new Date(date.toDateString())
+            controller.changeDeposit({depositId, description, depositDate, amount})
+            navigate("/deposits")
```

Comment:
I would wait with the navigation until the request is done, otherwise you cant be sure the deposit is already change and the user will see the new deposit on the deposites site...